tests/integration/pybaseball/datahelpers/test_caching.py

```diff
@@ -0,0 +1,30 @@
+from unittest.mock import patch
+from os import path
+import pandas as pd
+import pytest
+
+import pybaseball
+
+@pytest.mark.parametrize(
+    "cache_type", [(x) for x in pybaseball.datahelpers.caching.CacheType]
+)
+def test_cache(cache_type: pybaseball.datahelpers.caching.CacheType) -> None:
+    with patch('pybaseball.datahelpers.caching.cache_config',
+               pybaseball.datahelpers.caching.CacheConfig(enabled=True, cache_type=cache_type, profiling=True,
+                                                          cache_directory=path.join(pybaseball.datahelpers.caching.cache_config.cache_directory, '.pytest'))
+    ):
+        # Delete any existing data just in case
+        pybaseball.datahelpers.caching.bust_cache()
+
+        # Uncached read, cached write
+        result, _, write_time, _ = pybaseball.batting_stats(2019) # type: ignore
+
+        # Cached read, no write
+        result2, read_time2, _, file_size2 = pybaseball.batting_stats(2019) # type: ignore
```

Comment:
Where did the other parameters come from?

Reply:
If profiling is turned on, the cache wrapper will return these.